### ISSUE TITLE
Class fields cause false positive no-undef

### ISSUE THREAD

**Issue by ackvf** [eslint/typescript-eslint-parser#471]
**What version of TypeScript are you using?**
`2.7.2`, `2.8.1`

**What version of `typescript-eslint-parser` are you using?**

`15.0.0`

**What code were you trying to parse?**

```ts
class X {
  field = {} // [eslint] 'field' is not defined. (no-undef)
}
```

**What did you expect to happen?**
I did not expect any error

**What happened?**
I get `[eslint] 'field' is not defined. (no-undef)`

**Comment by todorone** [comment]
@ackvf DId You manage to resolve this issue?

**Comment by ackvf** [comment]
No, I had to turn the rule off. It still reports false errors with latest `16.0.1`. (`typescript 3.0.0-dev.20180626`)

**Comment by lostfictions** [comment]
I'm also experiencing this. Unfortunately, we're currently in the process of converting our codebase to TypeScript, so we have a mix of JS and TS files and it's not acceptable for us to turn `no-undef` off.

Unfortunately, seems like we'll simply need to revert to `babel-eslint` for the JS files and use TSLint instead for TypeScript.

**Comment by ackvf** [comment]
@lostfictions The undef property is still covered by tslint if you use both (like I do).

![image](https://user-images.githubusercontent.com/2313018/43582311-4e2ebcee-965c-11e8-87e8-99feec37ab6a.png)

**Comment by todorone** [comment]
@ackvf Thanks for the hint. I don't want to introduce `tslint` in my project so I just had to turn this rule off...

**Comment by timche** [comment]
Check `no-undef` with `tsc` instead:

1. Disable `no-undef` in ESLint
2. Turn on `"noUnusedLocals": true` instead in `tsconfig.json`
3. `tsc` will now error at not defined variables

**Comment by timche** [comment]
@lostfictions You are right, I'm confusing it with `no-unused-vars`. 

There are currently two rules I've disabled in ESLint:

- `no-undef`
- `no-unused-vars`

Both are causing false-positives in ESLint with TypeScript.

By deactivating them, we now let `tsc` check them: 

- `no-undef`: No configuration required
- `no-unused-vars`: `"noUnusedLocals": true` in `tsconfig.json`

**Comment by Gvozd** [comment]
@timche `no-unused-vars` can be replaced with `typescript/no-unused-vars` from `eslint-plugin-typescript`
